sdk/include/opentelemetry/sdk/logs/processor.h

```diff
@@ -25,21 +25,44 @@ namespace sdk
 {
 namespace logs
 {
+enum class ShutdownResult
+{
+  kSuccess = 0,
+  kFailure = 1,
+  kTimeout = 2
+};
 /**
- * This Log Processor is responsible for conversion of logs to exportable
- * representation and passing them to exporters.
+ * This Log Processor is responsible for the batching of log records
+ * and passing them to exporters.
  */
 class LogProcessor
 {
 public:
   virtual ~LogProcessor() = default;
 
+  /**
+   * OnReceive is responsible for batching every log record that is created by the SDK
+   * @param record a log record that has all the user data and injected data
+   */
   virtual void OnReceive(std::unique_ptr<opentelemetry::logs::LogRecord> &&record) noexcept = 0;
 
-  virtual void ForceFlush(
+  /**
+   * Exports all log records that have not yet been exported to the configured Exporter.
+   * @param timeout that the forceflush is required to finish within.
+   * A default timeout of 0 mean no timeout is applied.
```

Comment:
I don't have strong opinion here but would like to get more perspectives.

In .NET the idiomatic way of specifying **infinite** is to use `Timeout.Infinite` which is `-1`. Here goes what OpenTelemetry .NET did https://github.com/open-telemetry/opentelemetry-dotnet/blob/2785ffbbaa5487d54e4f3cefe1821fc18af86550/src/OpenTelemetry/BaseExporter.cs#L65.

I wonder what's the idiomatic way for C++. Should we treat 0 as "trigger the flush and return as quick as we can, don't wait at all", or "block indefinitely until the operation succeeded/failed"?

Reply:
I would agree with option # 2.

The current implementation currently follows [traces](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/include/opentelemetry/sdk/trace/simple_processor.h#L57), but I think we should do what is done in the Dotnet above, is actually better. We could use `std::chrono::microseconds::max()`, instead of `std::chrono::microseconds(0)` to indicate an "infinite" timeout. This returns the [largest duration possible](http://www.enseignement.polytechnique.fr/informatique/INF478/docs/Cpp/en/cpp/chrono/duration/max.html), so would not actually be "infinite" amount of time, so it should align with the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shutdown-2)'s requirement for `Shutdown` not to block indefinitely.

Reply:
I agree the default of 0 doesn't make a lot of sense, but I think neither does not enforcing a max timeout at all.

I am leaning towards a middle ground- I think we could use `std::chrono::microseconds::max()`, instead of `std::chrono::microseconds(0)` for the max timeout. This would return the [largest duration possible](http://www.enseignement.polytechnique.fr/informatique/INF478/docs/Cpp/en/cpp/chrono/duration/max.html), so would not actually be "infinite" amount of time so it would not ever block indefinitely as per the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shutdown-2). 

If that is in line with what you're thinking, I can make that change.

Reply:
No strong opinion either, but I think it is good to have a well defined default value how large it may be to ensure that we are not blocking indefinitely.

Reply:
Flagged in #418